File: calculator/src/main/java/com/turlir/calculator/member/ValuePrint.java
package com.turlir.calculator.member;

import com.turlir.calculator.converter.Printer;
import com.turlir.calculator.converter.Visual;

import java.math.BigDecimal;

class ValuePrint implements Visual {

    private final BigDecimal mValue;
    private final boolean isFloat;

    ValuePrint(BigDecimal value, boolean shouldFloat) {
        mValue = value;
        isFloat = shouldFloat;
    }

    @Override
    public void print(Printer chain) {
        chain.append(mValue);
        if (isFloat) chain.appendSeparator();
    }

    @Override
    public String toString() {
        return mValue.toString();
    }
}

File: calculator/src/main/java/com/turlir/calculator/member/ValuePrint.kt
package com.turlir.calculator.member

import com.turlir.calculator.converter.Printer
import com.turlir.calculator.converter.Visual

import java.math.BigDecimal

internal class ValuePrint(private val mValue: BigDecimal, private val isFloat: Boolean) : Visual {

    override fun print(chain: Printer) {
        chain.append(mValue)
        if (isFloat) chain.appendSeparator()
    }

    override fun toString() = mValue.toString()
}
